src/result-async.ts

```diff
@@ -0,0 +1,123 @@
+import { Result, Ok, Err } from './'
+import { isPromise } from './'
+
+export class ResultAsync<T, E> {
+  private _promise: Promise<Result<T, E>>
+
+  constructor(res: Promise<Result<T, E>>) {
+    this._promise = res
+  }
+
+  static fromPromise<T, E>(promise: Promise<T>): ResultAsync<T, E> {
+    return new ResultAsync(promise.then((value: T) => new Ok<T, E>(value)))
```

Comment:
What happens if calling `promise.then` results in a rejected promise? It would be odd if neverthrow started throwing unhandled exceptions 🙈 !
  - What if there was a `catch` here that returned an `Err`?

Reply:
You are totally right and I actually had already modified this function in another branch.
What I propose is to offer a second argument to `fromPromise` to handle the error case. This second argument should be a function which receives whatever is thrown by the promise and returns an `E` type value.

It looks like this now:
```typescript
  static fromPromise<T, E>(promise: Promise<T>, errorFn?: (e: unknown) => E): ResultAsync<T, E> {
    let newPromise: Promise<Result<T, E>> = promise.then((value: T) => new Ok(value))
    if (errorFn) {
      newPromise = newPromise.catch(e => new Err<T, E>(errorFn(e)))
    }
    return new ResultAsync(newPromise)
  }
```